Read `cloudflare_record` resources where type is `DNSKEY`

### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.5.0
on windows_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v4.8.0


### Affected resource(s)

`cloudflare_record`

### Terraform configuration files

```hcl
terraform {
  required_providers {
    cloudflare = {
      source  = "cloudflare/cloudflare"
      version = "~> 4"
    }
  }
}
resource "cloudflare_record" "terraform_managed_resource_cde2a1432fd4c975f921e6ea5177b5a9" {
  name    = "<name>"
  proxied = false
  ttl     = 1
  type    = "DNSKEY"
  zone_id = var.zone_id
  data {
    algorithm  = 2
    flags      = 2371
    protocol   = 13
    public_key = "<key>"
  }
}
```


### Link to debug output

https://gist.github.com/Cyb3r-Jak3/dbfe4842ef677254efbbefae8be7b213

### Panic output

```
Stack trace from the terraform-provider-cloudflare_v4.8.0.exe plugin:

panic: interface conversion: interface {} is float64, not string

goroutine 95 [running]:
github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider.transformToCloudflareDNSData({0xc00012a108, 0x6}, {0xc00012a158?, 0x9?}, {0x139a600, 0xc00012a168})
        github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider/resource_cloudflare_record.go:436 +0x505
github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider.resourceCloudflareRecordRead({0x16ecca0, 0xc00045d830}, 0xc00064ea80, {0x15291e0?, 0xc000174780})
        github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider/resource_cloudflare_record.go:221 +0x855
github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider.resourceCloudflareRecordImport({0x16ecca0, 0xc00045d830}, 0xc00064ea80, {0x15291e0?, 0xc000174780?})
        github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider/resource_cloudflare_record.go:396 +0x738
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).ImportState(0xc000090ea0, {0x16ecca0, 0xc00045d830}, 0xc00086f390, {0xc000478000, 0x41})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/provider.go:385 +0x1ac
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ImportResourceState(0xc000009338, {0x16ecca0?, 0xc00045d710?}, 0xc0006100c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1088 +0xf4
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ImportResourceState({{0x16f29f0?, 0xc000009338?}}, {0x16ecca0?, 0xc00045d710?}, 0xc0006100a0?)
        github.com/hashicorp/terraform-plugin-mux@v0.10.0/tf5to6server/tf5to6server.go:67 +0xd0
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.muxServer.ImportResourceState({0xc0003b31a0, 0xc0003b3200, {0xc000611480, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
        github.com/hashicorp/terraform-plugin-mux@v0.10.0/tf6muxserver/mux_server_ImportResourceState.go:27 +0x102
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ImportResourceState(0xc0004cb0e0, {0x16ecca0?, 0xc00045cab0?}, 0xc00046c0f0)
        github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov6/tf6server/server.go:846 +0x1f6
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ImportResourceState_Handler({0x14f9b20?, 0xc0004cb0e0}, {0x16ecca0, 0xc00045cab0}, 0xc0001ae070, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:437 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004b85a0, {0x16f1978, 0xc0005841a0}, 0xc0004de000, 0xc000433a10, 0x1cf8278, 0x0)
        google.golang.org/grpc@v1.54.0/server.go:1345 +0xdf3
google.golang.org/grpc.(*Server).handleStream(0xc0004b85a0, {0x16f1978, 0xc0005841a0}, 0xc0004de000, 0x0)
        google.golang.org/grpc@v1.54.0/server.go:1722 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.54.0/server.go:966 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.54.0/server.go:964 +0x28a

Error: The terraform-provider-cloudflare_v4.8.0.exe plugin crashed!
```

### Expected output

Able to import DNSKEY record

### Actual output

Plugin crashes

### Steps to reproduce

1. Have a DNSKEY record in dashboard
2. Use `cf-terraforming` to generate the resource block
3. Attempt to import.

### Additional factoids

Record was generated using cf-terraform

### References

Just need to move `strings.ToUpper(recordType) == "DNSKEY":` to where the CAA type is

https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/internal/sdkv2provider/resource_cloudflare_record.go#L432-L448

--- a/.changelog/2568.txt
+++ b/.changelog/2568.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/cloudflare_record: fix importing of DNSKEY record types
+```

--- a/internal/sdkv2provider/resource_cloudflare_record.go
+++ b/internal/sdkv2provider/resource_cloudflare_record.go
@@ -431,12 +431,11 @@ func transformToCloudflareDNSData(recordType string, id string, value interface{
 	switch {
 	case id == "flags":
 		switch {
-		case strings.ToUpper(recordType) == "SRV",
-			strings.ToUpper(recordType) == "DNSKEY":
+		case strings.ToUpper(recordType) == "SRV":
 			newValue, err = value.(string), nil
 		case strings.ToUpper(recordType) == "NAPTR":
 			newValue, err = value.(string), nil
-		case strings.ToUpper(recordType) == "CAA":
+		case strings.ToUpper(recordType) == "CAA", strings.ToUpper(recordType) == "DNSKEY":
 			// this is required because "flags" is shared however, it comes from
 			// the API as a float64 but the Terraform internal type is string 😢.
 			switch value.(type) {

--- a/internal/sdkv2provider/resource_cloudflare_record_test.go
+++ b/internal/sdkv2provider/resource_cloudflare_record_test.go
@@ -563,6 +563,35 @@ func TestAccCloudflareRecord_MXNull(t *testing.T) {
 	})
 }
 
+func TestAccCloudflareRecord_DNSKEY(t *testing.T) {
+	t.Parallel()
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+	rnd := generateRandomResourceName()
+	name := fmt.Sprintf("cloudflare_record.%s", rnd)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccPreCheckDomain(t)
+		},
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckCloudflareRecordDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareRecordDNSKEY(zoneID, domain),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "name", domain),
+					resource.TestCheckResourceAttr(name, "type", "DNSKEY"),
+					resource.TestCheckResourceAttr(name, "data.0.flags", "257"),
+					resource.TestCheckResourceAttr(name, "data.0.protocol", "13"),
+					resource.TestCheckResourceAttr(name, "data.0.algorithm", "2"),
+					resource.TestCheckResourceAttr(name, "data.0.public_key", "mdsswUyr3DPW132mOi8V9xESWE8jTo0dxCjjnopKl+GqJxpVXckHAeF+KkxLbxILfDLUT0rAK9iUzy1L53eKGQ=="),
+				),
+			},
+		},
+	})
+}
+
 func TestAccCloudflareRecord_ClearTags(t *testing.T) {
 	t.Parallel()
 	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
@@ -942,4 +971,21 @@ resource "cloudflare_record" "%[3]s" {
 	type = "A"
 	ttl = 3600
 }`, zoneID, name, rnd)
+}
+
+func testAccCheckCloudflareRecordDNSKEY(zoneID, name string) string {
+	return fmt.Sprintf(`
+	 resource "cloudflare_record" "dnskey" {
+ 		zone_id = "%[1]s"
+	   	name    = "%[2]s"
+	   	type    = "DNSKEY"
+	  
+	   	data {
+			algorithm  = 2
+		 	flags      = 2371
+		 	protocol   = 13
+		 	public_key = "mdsswUyr3DPW132mOi8V9xESWE8jTo0dxCjjnopKl+GqJxpVXckHAeF+KkxLbxILfDLUT0rAK9iUzy1L53eKGQ=="
+	   }
+	 }
+`, zoneID, name)
 }

--- a/internal/sdkv2provider/resource_cloudflare_record_test.go
+++ b/internal/sdkv2provider/resource_cloudflare_record_test.go
@@ -564,6 +564,8 @@ func TestAccCloudflareRecord_MXNull(t *testing.T) {
 }
 
 func TestAccCloudflareRecord_DNSKEY(t *testing.T) {
+	skipForDefaultZone(t, "Pending automating setup from https://developers.cloudflare.com/dns/dnssec/multi-signer-dnssec/.")
+
 	t.Parallel()
 	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
 	rnd := generateRandomResourceName()
@@ -979,7 +981,7 @@ func testAccCheckCloudflareRecordDNSKEY(zoneID, name string) string {
  		zone_id = "%[1]s"
 	   	name    = "%[2]s"
 	   	type    = "DNSKEY"
-	  
+
 	   	data {
 			algorithm  = 2
 		 	flags      = 2371

